File: src/main/scala/com/fyc/dataStream/wordcount.scala
package com.fyc.dataStream

import org.apache.flink.configuration.{ConfigConstants, Configuration}
import org.apache.flink.streaming.api.scala._

//必须先监听9000端口 cmd到netcat目录下  执行nc64.exe -l -p 9000
object wordcount {
  def main(args: Array[String]): Unit = {

    val configuration = new Configuration()
    configuration.setBoolean(ConfigConstants.LOCAL_START_WEBSERVER,true)
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(configuration)
    val ds: DataStream[String] = env.socketTextStream("localhost",9000)
    val count: DataStream[(String, Int)] = ds.flatMap(_.split(" "))
      .map((_, 1))
      .keyBy(0)
      //.timeWindow(Time.seconds(5),Time.seconds(5))
      .sum(1)
    count.print().setParallelism(1)
    env.execute("word count")







  }
}

File: src/main/java/com/fyc/dataStream/wordCount.java
/**
 * FileName: wordCount
 * Author:   DFJX
 * Date:     2019/11/27 16:37
 * Description:
 * History:
 * <author>          <time>          <version>          <desc>
 * 作者姓名           修改时间           版本号              描述
 */
package com.fyc.dataStream;


import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.configuration.ConfigConstants;
import org.apache.flink.configuration.Configuration;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.api.windowing.time.Time;
import org.apache.flink.util.Collector;

/**
 * 〈一句话功能简述〉<br> 
 * 〈〉
 *
 * @author DFJX
 * @create 2019/11/27
 * @since 1.0.0
 */
public class wordCount {
    public static void main(String[] args) throws Exception {
        Configuration config = new Configuration();
        config.setBoolean(ConfigConstants.LOCAL_START_WEBSERVER, true);
        StreamExecutionEnvironment env = StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(config);
        DataStreamSource<String> socketTextStream = env.socketTextStream("localhost", 9000);
        SingleOutputStreamOperator<Tuple2<String, Integer>> sum = socketTextStream.flatMap(new FlatMapFunction<String, Tuple2<String, Integer>>() {
            public void flatMap(String value, Collector<Tuple2<String, Integer>> out) throws Exception {
                String[] s = value.split(" ");
                for (String s1 : s) {
                    out.collect(new Tuple2<String, Integer>(s1, 1));
                }
            }
        }).keyBy(0).timeWindow(Time.seconds(1)).sum(1);
        sum.print().setParallelism(1);
        env.execute("java stream wordcount");

    }

}
